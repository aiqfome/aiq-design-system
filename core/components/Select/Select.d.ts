import React from 'react';
import { Props as BoxProps } from '../Box';
export interface Props extends BoxProps {
    label?: string;
    items?: Array<string | {
        id: any;
        name: any;
        select?: any;
    }>;
    isOpen?: boolean;
    variant?: 'outlined';
    prefix?: any;
    placeholder?: string;
    handleSelectedItemChange?: (item: any) => void;
    onChangeTextInput?: (text: string) => void;
    selectedItem?: any;
    autoComplete?: boolean;
    sufix?: any;
    isLoading?: boolean;
    errorMessage?: string;
    errorForm?: boolean;
    isFetchable?: boolean;
    inputProps?: any;
    loadingMessage?: string;
    emptyMessage?: string;
    defaultValue?: any;
    clearOnSelect?: boolean;
    isDependent?: boolean;
    dependentMessage?: string;
}
export declare const Select: React.FC<Props>;
//# sourceMappingURL=Select.d.ts.map