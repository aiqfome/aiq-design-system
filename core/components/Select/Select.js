"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Select = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const SelectFetchable_1 = require("./SelectFetchable");
const SelectStatic_1 = require("./SelectStatic");
exports.Select = (_a) => {
    var { isFetchable } = _a, props = __rest(_a, ["isFetchable"]);
    if (isFetchable) {
        return react_1.default.createElement(SelectFetchable_1.SelectFetchable, Object.assign({}, props));
    }
    return react_1.default.createElement(SelectStatic_1.SelectStatic, Object.assign({}, props));
};
exports.Select.propTypes = {
    isFetchable: prop_types_1.default.bool
};
