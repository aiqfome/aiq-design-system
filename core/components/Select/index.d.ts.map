{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Select/index.tsx"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAA"}