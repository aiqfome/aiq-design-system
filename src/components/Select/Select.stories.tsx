import React, { useState } from 'react'
import { withKnobs, text, object } from '@storybook/addon-knobs'
import { Select } from './Select'
import { Flex } from '../Flex'

import { MdSearch } from 'react-icons/md'

export default {
  component: Select,
  title: 'Select',
  decorators: [withKnobs]
}

export const Basic: React.FC = () => {
  const itens = ['React', 'CSS', 'PHP', 'HTML']

  function handleClickItemSelect(item) {
    console.log(item)
  }

  return (
    <Select
      handleSelectedItemChange={handleClickItemSelect}
      placeholder={text('label', 'aiq-design-system')}
      items={object('items', itens)}
    />
  )
}

export const Outlined: React.FC = () => {
  const itens = ['React', 'CSS', 'PHP', 'HTML']

  return (
    <Select
      variant='outlined'
      label={text('label', 'aiq-design-system')}
      items={object('items', itens)}
    />
  )
}

export const AutocompleteFalse: React.FC = () => {
  const itens = [
    { name: 'React', id: 1 },
    { name: 'CSS', id: 2 },
    { name: 'PHP', id: 3 },
    { name: 'HTML', id: 4 }
  ]
  const [selectedItem, setSelectedItem] = useState(itens[0])
  return (
    <Flex variant='fullCentralized'>
      <Select
        mr='32px'
        autoComplete={false}
        variant='outlined'
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
      />
      <Select
        selectedItem={selectedItem}
        autoComplete={false}
        handleSelectedItemChange={({ selectedItem }) =>
          setSelectedItem(selectedItem)
        }
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
      />
    </Flex>
  )
}

export const WithSufix: React.FC = () => {
  const itens = [
    { name: 'React', id: 1 },
    { name: 'CSS', id: 2 },
    { name: 'PHP', id: 3 },
    { name: 'HTML', id: 4 }
  ]
  const [selectedItem, setSelectedItem] = useState(itens[0])
  return (
    <Flex variant='fullCentralized'>
      <Select
        mr='32px'
        variant='outlined'
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
        sufix={<MdSearch />}
      />
      <Select
        selectedItem={selectedItem}
        handleSelectedItemChange={({ selectedItem }) =>
          setSelectedItem(selectedItem)
        }
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
        sufix={<MdSearch />}
      />
    </Flex>
  )
}

export const Loading: React.FC = () => {
  const itens = [
    { name: 'React', id: 1 },
    { name: 'CSS', id: 2 },
    { name: 'PHP', id: 3 },
    { name: 'HTML', id: 4 }
  ]
  const [selectedItem, setSelectedItem] = useState(itens[0])
  return (
    <Flex variant='fullCentralized'>
      <Select
        mr='32px'
        variant='outlined'
        isLoading={true}
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
        sufix={<MdSearch />}
      />
      <Select
        selectedItem={selectedItem}
        isLoading={true}
        handleSelectedItemChange={({ selectedItem }) =>
          setSelectedItem(selectedItem)
        }
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
        sufix={<MdSearch />}
      />
    </Flex>
  )
}
