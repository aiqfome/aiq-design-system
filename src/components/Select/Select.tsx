import React, { useState } from 'react'
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import { useCombobox } from 'downshift'
import { IoIosArrowDown } from 'react-icons/io'

import { Box, Props as BoxPros } from '../Box'
import { Input } from '../Input'
import { Button, Props as ButtonProps } from '../Button'

export interface Props extends BoxPros {
  label?: string
  items?: Array<string | { value: any; label: any }>
  isOpen?: boolean
  variant?: 'outlined'
  prefix?: any
  placeholder?: string
  handleSelectedItemChange?: (item: any) => void
  onChangeTextInput?: (text: string) => void
  selectedItem?: any
  autoComplete?: boolean
  sufix?: any
}

const Container = styled(Box)<Props>`
  position: relative;

  ul {
    box-shadow: 0px 3px 6px #00000029;
    background: ${({ theme }) => theme.colors.white};
    border: 1px solid ${({ theme }) => theme.colors.lightGrey};
    list-style-type: none;
    position: absolute;
    top: ${({ variant }) => (variant === 'outlined' ? '39px' : '38px')};
    overflow: hidden;
    z-index: 1;
    width: 100%;
    padding: 0;
    margin: 0;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;

    li {
      cursor: pointer;
      padding: 6px 12px;
    }

    ${({ isOpen }) =>
      !isOpen &&
      css`
        display: none;
      `}
  }
`
interface ItemProps {
  highlighted?: boolean
}

const Item = styled.li<ItemProps>`
  background: ${({ highlighted, theme }) =>
    highlighted ? theme.colors.primaryLight : '#fff'};
`

interface ButtonStyledProps extends ButtonProps {
  variantSelect?: any
}

const ButtonStyled = styled(Button)<ButtonStyledProps>`
  position: absolute;
  top: ${({ variantSelect }) =>
    variantSelect === 'outlined' ? '13px' : '12px'};
  right: 14px;
`

export const Select: React.FC<Props> = ({
  label,
  variant,
  items = [],
  placeholder,
  selectedItem,
  autoComplete = true,
  sufix,
  handleSelectedItemChange = () => {
    // do nothing.
  },
  onChangeTextInput = () => {
    // do nothing.
  },
  prefix,
  ...props
}) => {
  const [inputItems, setInputItems] = useState(items)

  const { backgroundColor, border, width, maxWidth } = props
  const boxStyled = {
    backgroundColor,
    border,
    width,
    maxWidth
  }

  const {
    isOpen,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    highlightedIndex,
    getToggleButtonProps,
    openMenu,
    getItemProps
  } = useCombobox({
    onSelectedItemChange: handleSelectedItemChange,
    items: inputItems,
    selectedItem,
    itemToString: item => (typeof item === 'string' ? item : item.label),
    onInputValueChange: ({ inputValue = '' }) => {
      if (autoComplete) {
        onChangeTextInput(inputValue)
        setInputItems(
          items.filter(item => {
            const label = typeof item === 'string' ? item : item.label
            return label.toLowerCase().startsWith(inputValue.toLowerCase())
          })
        )
      }
    }
  })

  function handleClickInput() {
    if (!isOpen) {
      openMenu()
    }
  }

  return (
    <Container isOpen={isOpen} variant={variant} {...props}>
      <ul {...getMenuProps()}>
        {isOpen &&
          inputItems &&
          inputItems.length > 0 &&
          inputItems.map((item, index) => (
            <Item
              key={`${index}`}
              highlighted={highlightedIndex === index}
              {...getItemProps({ item, index })}
            >
              {typeof item === 'string' ? item : item.label}
            </Item>
          ))}
      </ul>

      <Box refBox={getComboboxProps().ref}>
        <Input
          onChange={getInputProps().onChange}
          onBlur={getInputProps().onBlur}
          onKeyDown={getInputProps().onKeyDown}
          onClick={handleClickInput}
          value={getInputProps().value}
          inputRef={getInputProps().ref}
          variant={variant}
          label={label}
          readOnly={!autoComplete}
          prefix={prefix}
          placeholder={placeholder}
          {...boxStyled}
        />
        {inputItems && (
          <ButtonStyled
            type='button'
            palette='primary'
            mr={5}
            variantSelect={variant}
            onClick={getToggleButtonProps().onClick}
            refButton={getToggleButtonProps().ref}
            aria-label='toggle menu'
          >
            {sufix || <IoIosArrowDown />}
          </ButtonStyled>
        )}
      </Box>
    </Container>
  )
}

Select.propTypes = {
  label: PropTypes.string,
  items: PropTypes.array,
  isOpen: PropTypes.bool,
  variant: PropTypes.oneOf(['outlined']),
  prefix: PropTypes.any,
  placeholder: PropTypes.string,
  handleSelectedItemChange: PropTypes.func,
  onChangeTextInput: PropTypes.func,
  selectedItem: PropTypes.any,
  autoComplete: PropTypes.bool,
  backgroundColor: PropTypes.any,
  border: PropTypes.any,
  width: PropTypes.any,
  maxWidth: PropTypes.any,
  sufix: PropTypes.any
}

Select.defaultProps = {
  items: []
}
