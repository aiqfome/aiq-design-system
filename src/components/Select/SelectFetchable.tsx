import { useCombobox } from 'downshift'
import React, { useEffect, useMemo, useState } from 'react'
import { IoIosArrowDown } from 'react-icons/io'
import styled, { css } from 'styled-components'

import { Box, Props as BoxPros } from '../Box'
import { Button, Props as ButtonProps } from '../Button'
import { Input } from '../Input'
import { Loading } from '../Loading'
import { Text } from '../Text'

export type Props = BoxPros & {
  label?: string
  internalLabel?: string
  items?: Array<string | { id: any; name: any; select?: any }>
  isOpen?: boolean
  variant?: 'outlined'
  prefix?: any
  placeholder?: string
  handleSelectedItemChange?: (item: any) => void
  onChangeTextInput?: (text: string) => void
  handleClearSelection?: (clear: '') => void
  selectedItem?: any
  autoComplete?: boolean
  sufix?: any
  isLoading?: boolean
  errorMessage?: string
  errorForm?: boolean
  onChange?: any
  loadingMessage?: string
  emptyMessage?: string
  inputProps?: any
  isDependent?: boolean
  disabled?: boolean
  dependentMessage?: string
}

const Container = styled(Box)<Props>`
  position: relative;

  input {
    cursor: auto;
    padding-right: 30px;
  }

  ul {
    box-shadow: 0px 3px 6px #00000029;
    background: ${({ theme }) => theme.colors.white};
    border: 1px solid ${({ theme }) => theme.colors.lightGrey};
    list-style-type: none;
    position: absolute;
    top: ${({ variant }) => (variant === 'outlined' ? '39px' : '38px')};
    overflow: hidden;
    z-index: 1;
    width: 100%;
    padding: 0;
    margin: 0;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
    max-height: 300px;
    overflow-y: auto;

    li {
      cursor: pointer;
      padding: 6px 12px;
      word-wrap: break-word;
      hyphens: auto;
    }

    ${({ isOpen }) =>
      !isOpen &&
      css`
        display: none;
      `}
  }
`
interface ItemProps {
  highlighted?: boolean
}

const Item = styled.li<ItemProps>`
  background: ${({ highlighted, theme }) =>
    highlighted ? theme.colors.primaryLight : '#fff'};
`

interface VariantSelect extends ButtonProps {
  variantSelect?: any
}

const ButtonStyled = styled(Button)<VariantSelect>`
  position: absolute;
  top: ${({ variantSelect }) =>
    variantSelect === 'outlined' ? '13px' : '12px'};
  right: 14px;
`

const LoadingBox = styled(Box)<VariantSelect>`
  position: absolute;
  top: ${({ variantSelect }) =>
    variantSelect === 'outlined' ? '13px' : '12px'};
  right: 14px;
`
export const SelectFetchable = React.forwardRef<HTMLDivElement, Props>(
  (
    {
      disabled,
      label,
      internalLabel,
      variant,
      items = [],
      placeholder,
      selectedItem,
      autoComplete = true,
      sufix,
      isLoading,
      errorMessage,
      errorForm,
      onChange,
      isDependent = false,
      dependentMessage = 'este campo tem alguma dependência',
      loadingMessage = 'carregando...',
      emptyMessage = 'item não encontrado',
      handleSelectedItemChange = () => {
        // do nothing.
      },
      onChangeTextInput = () => {
        // do nothing.
      },
      handleClearSelection = () => {
        // do nothing.
      },
      prefix,
      inputProps,
      ...props
    },
    ref
  ) => {
    const [inputItems, setInputItems] = useState(items)

    useEffect(() => setInputItems(items), [items])

    const { backgroundColor, border, width, maxWidth } = props
    const boxStyled = {
      backgroundColor,
      border,
      width,
      maxWidth,
      ...inputProps
    }

    const {
      isOpen,
      getMenuProps,
      getInputProps,
      getComboboxProps,
      highlightedIndex,
      getToggleButtonProps,
      openMenu,
      getItemProps,
      setInputValue
    } = useCombobox({
      onSelectedItemChange: handleSelectedItemChange,
      items: isDependent ? [] : inputItems,
      selectedItem,
      itemToString: item => (typeof item === 'string' ? item : item.name),
      onInputValueChange: ({ inputValue = '' }) => {
        if (autoComplete && typeof inputValue === 'string') {
          onChangeTextInput(inputValue)
          onChange(inputValue)

          setInputItems(
            items.filter(item => {
              const name =
                typeof item === 'string' ? item : item.select || item.name
              return name.toLowerCase().startsWith(inputValue.toLowerCase())
            })
          )
        }
      }
    })

    function handleClickInput() {
      setInputItems(items)
      if (!isOpen) {
        setInputValue('')
        handleClearSelection('')
        openMenu()
      }
    }

    const completePrefix = useMemo(() => {
      if (!prefix && !internalLabel) return null

      return (
        <div style={{ marginRight: internalLabel && '-12px', width: 'max-content' }}>
          {prefix}
          <Text ml={prefix && '4px'} color='darkGrey' fontSize='small' >{internalLabel}</Text>
        </div>
      )
    }, [])

    return (
      <Container
        isOpen={isOpen && !disabled}
        variant={variant}
        data-testid='select-fechable'
        ref={ref}
        {...props}
      >
        <ul {...getMenuProps()}>
          {internalLabel && 
            <div style={{ marginTop: '8px', marginBottom: '2px' }}>
              <Text p='12px' color='darkGrey' fontSize={12} fontWeight='medium'>
                {internalLabel}
              </Text>
            </div>
          }
          {isOpen &&
            inputItems &&
            !isLoading &&
            !disabled &&
            !isDependent &&
            inputItems.length > 0 &&
            inputItems.map((item, index) => (
              <Item
                key={index}
                data-testid='select-item'
                highlighted={highlightedIndex === index}
                {...getItemProps({ item, index })}
              >
                {typeof item === 'string' ? item : item.name}
              </Item>
            ))}

          {isOpen && isLoading && !disabled && !isDependent && (
            <Item>{loadingMessage}</Item>
          )}

          {isOpen &&
            inputItems &&
            !isLoading &&
            !disabled &&
            !isDependent &&
            inputItems.length === 0 && <Item>{emptyMessage}</Item>}

          {isDependent && <Item>{dependentMessage}</Item>}
        </ul>

        <Box ref={getComboboxProps().ref}>
          <Input
            disabled={disabled}
            onChange={getInputProps().onChange}
            onBlur={getInputProps().onBlur}
            onKeyDown={getInputProps().onKeyDown}
            onClick={handleClickInput}
            value={getInputProps().value}
            inputRef={getInputProps().ref}
            variant={variant}
            label={label}
            errorMessage={errorMessage}
            errorForm={errorForm}
            readOnly={!autoComplete}
            prefix={completePrefix}
            placeholder={placeholder}
            nativeAutoComplete='disabled'
            {...boxStyled}
          />
          {isLoading && (
            <LoadingBox>
              <Loading size='small' />
            </LoadingBox>
          )}
          {inputItems && !isLoading && (
            <ButtonStyled
              disabled={disabled}
              type='button'
              palette='primary'
              variantSelect={variant}
              onClick={getToggleButtonProps().onClick}
              refButton={getToggleButtonProps().ref}
              aria-label='toggle menu'
            >
              {sufix || <IoIosArrowDown />}
            </ButtonStyled>
          )}
        </Box>
      </Container>
    )
  }
)

SelectFetchable.defaultProps = {
  items: []
}
