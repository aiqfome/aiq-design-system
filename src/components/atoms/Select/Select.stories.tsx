import React, { useState } from 'react'
import { withKnobs, text, object } from '@storybook/addon-knobs'
import { Select } from './Select'
import { Flex } from '../Flex'

export default {
  component: Select,
  title: 'atoms/Select',
  decorators: [withKnobs]
}

export const Basic: React.FC = () => {
  const itens = ['React', 'CSS', 'PHP', 'HTML']

  function handleClickItemSelect(item) {
    console.log(item)
  }

  return (
    <Select
      handleSelectedItemChange={handleClickItemSelect}
      placeholder={text('label', 'aiq-design-system')}
      items={object('items', itens)}
    />
  )
}

export const Outlined: React.FC = () => {
  const itens = ['React', 'CSS', 'PHP', 'HTML']

  return (
    <Select
      variant='outlined'
      label={text('label', 'aiq-design-system')}
      items={object('items', itens)}
    />
  )
}

export const AutocompleteFalse: React.FC = () => {
  const itens = ['React', 'CSS', 'PHP', 'HTML']
  const [selectedItem, setSelectedItem] = useState(itens[0])
  return (
    <Flex variant='fullCentralized'>
      <Select
        mr='32px'
        autoComplete={false}
        variant='outlined'
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
      />
      <Select
        autoComplete={false}
        selectedItem={selectedItem}
        handleSelectedItemChange={({ selectedItem }) =>
          setSelectedItem(selectedItem)
        }
        label={text('label', 'aiq-design-system')}
        items={object('items', itens)}
      />
    </Flex>
  )
}
