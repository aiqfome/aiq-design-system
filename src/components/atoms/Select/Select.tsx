import React, { useState } from 'react'
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import { useCombobox } from 'downshift'
import { IoIosArrowDown } from 'react-icons/io'

import { Box } from '../Box'
import { Input } from '../Input'
import { Button, Props as ButtonProps } from '../Button'

export interface Props {
  label?: string
  items?: string[]
  isOpen?: boolean
  variant?: 'outlined'
  prefix?: any
  placeholder?: string
  handleSelectedItemChange?: (item: any) => void
  selectedItem?: any

  backgroundColor?: any
  border?: any
  width?: any
  maxWidth?: any
}

const Container = styled(Box)<Props>`
  position: relative;
  ul {
    background: ${({ theme }) => theme.colors.white};
    border: 1px solid ${({ theme }) => theme.colors.lightGrey};
    padding: 10px;
    list-style-type: none;
    position: absolute;
    top: ${({ variant }) => (variant === 'outlined' ? '39px' : '38px')};
    overflow: hidden;
    z-index: 1;
    width: 100%;

    li {
      cursor: pointer;
      padding: 8px;
    }

    ${({ isOpen }) =>
      !isOpen &&
      css`
        display: none;
      `}
  }
`

interface ButtonStyledProps extends ButtonProps {
  variantSelect?: any
}

const ButtonStyled = styled(Button)<ButtonStyledProps>`
  position: absolute;
  top: ${({ variantSelect }) =>
    variantSelect === 'outlined' ? '15px' : '12px'};
  right: 14px;
`

export const Select: React.FC<Props> = ({
  label,
  variant,
  items = [],
  placeholder,
  selectedItem,
  handleSelectedItemChange,
  prefix,
  ...props
}) => {
  const [inputItems, setInputItems] = useState(items)

  const { backgroundColor, border, width, maxWidth } = props
  const boxStyled = {
    backgroundColor,
    border,
    width,
    maxWidth
  }

  const {
    isOpen,
    getMenuProps,
    getInputProps,
    getComboboxProps,
    highlightedIndex,
    getToggleButtonProps,
    getItemProps
  } = useCombobox({
    items: inputItems,
    selectedItem,
    onSelectedItemChange: handleSelectedItemChange,
    onInputValueChange: ({ inputValue = '' }) => {
      setInputItems(
        items.filter(item =>
          item.toLowerCase().startsWith(inputValue.toLowerCase())
        )
      )
    }
  })

  return (
    <Container isOpen={isOpen} variant={variant} {...props}>
      <ul {...getMenuProps()}>
        {isOpen &&
          inputItems &&
          inputItems.length > 0 &&
          inputItems.map((item, index) => (
            <li
              style={
                highlightedIndex === index ? { backgroundColor: '#bde4ff' } : {}
              }
              key={`${item}${index}`}
              {...getItemProps({ item, index })}
            >
              {item}
            </li>
          ))}
      </ul>

      <Box refBox={getComboboxProps().ref}>
        <Input
          onChange={getInputProps().onChange}
          onBlur={getInputProps().onBlur}
          onKeyDown={getInputProps().onKeyDown}
          value={getInputProps().value}
          inputRef={getInputProps().ref}
          variant={variant}
          label={label}
          prefix={prefix}
          placeholder={placeholder}
          {...boxStyled}
        />
        {inputItems && (
          <ButtonStyled
            palette='primary'
            mr={5}
            variantSelect={variant}
            refButton={getToggleButtonProps().ref}
            onClick={getToggleButtonProps().onClick}
            aria-label='toggle menu'
          >
            <IoIosArrowDown />
          </ButtonStyled>
        )}
      </Box>
    </Container>
  )
}

Select.propTypes = {
  label: PropTypes.string,
  items: PropTypes.array,
  isOpen: PropTypes.bool,
  variant: PropTypes.oneOf(['outlined']),
  prefix: PropTypes.any,
  placeholder: PropTypes.string,
  handleSelectedItemChange: PropTypes.func,
  selectedItem: PropTypes.any,

  backgroundColor: PropTypes.any,
  border: PropTypes.any,
  width: PropTypes.any,
  maxWidth: PropTypes.any
}

Select.defaultProps = {
  items: []
}
