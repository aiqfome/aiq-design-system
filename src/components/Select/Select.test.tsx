import React from 'react'

import { Select } from './Select'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<Select />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
