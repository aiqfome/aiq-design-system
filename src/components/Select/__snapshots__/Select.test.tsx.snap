// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxjAm giuoCA sc-fzpans fJzagG"
>
  <ul
    aria-labelledby="downshift-1-label"
    id="downshift-1-menu"
    onMouseLeave={[Function]}
    role="listbox"
  />
  <div
    className="sc-AxjAm giuoCA"
  >
    <div
      className="sc-AxirZ cdhGFS"
    >
      <input
        className="sc-Axmtr lRNal"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value=""
      />
    </div>
    <button
      aria-label="toggle menu"
      className="sc-AxhUy izzhmx sc-fznyAO iCMCtW"
      onClick={[Function]}
    >
      <span
        className="sc-AxhCb iVPPkG"
        cursor="pointer"
        fontSize="medium"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
